wasmbinding/message_plugin.go

```diff
@@ -399,60 +401,102 @@ func (m *CustomMessenger) performSubmitAdminProposal(ctx sdk.Context, contractAd
+		"from_address", contractAddr.String(),
+		"creator", contractAddr.String(),
+	)
+
+	return response, nil
+}
+
+func (m *CustomMessenger) performSubmitAdminProposal(ctx sdk.Context, contractAddr sdk.AccAddress, submitAdminProposal *bindings.SubmitAdminProposal) (*admintypes.MsgSubmitProposalResponse, error) {
+	proposal := submitAdminProposal.AdminProposal
+	authority := authtypes.NewModuleAddress(admintypes.ModuleName).String()
+
+	err := m.validateProposalQty(&proposal)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to validate proposal quantity")
+	}
+
+	var msg *admintypes.MsgSubmitProposal
+	var sdkMsgs []sdk.Msg
+	var sdkMsg sdk.Msg
+
+	if proposal.SoftwareUpgradeProposal != nil {
+		p := proposal.SoftwareUpgradeProposal
+		sdkMsg = &softwareUpgrade.MsgSoftwareUpgrade{
+			Authority: authority,
+			Plan: softwareUpgrade.Plan{
+				Name:   p.Plan.Name,
+				Height: p.Plan.Height,
+				Info:   p.Plan.Info,
+			},
+		}
+	}
+
+	if proposal.CancelSoftwareUpgradeProposal != nil {
+		sdkMsg = &softwareUpgrade.MsgCancelUpgrade{Authority: authority}
+	}
+
 	if proposal.PinCodesProposal != nil {
 		p := proposal.PinCodesProposal
-		err := msg.SetContent(&wasmtypes.PinCodesProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			CodeIDs:     p.CodeIDs,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on PinCodesProposal")
-		}
+		sdkMsg = &wasmtypes.MsgPinCodes{Authority: authority, CodeIDs: p.CodeIDs}
 	}
 
 	if proposal.UnpinCodesProposal != nil {
 		p := proposal.UnpinCodesProposal
-		err := msg.SetContent(&wasmtypes.UnpinCodesProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			CodeIDs:     p.CodeIDs,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on UnpinCodesProposal")
-		}
+		sdkMsg = &wasmtypes.MsgUnpinCodes{Authority: authority, CodeIDs: p.CodeIDs}
 	}
 
 	if proposal.UpdateAdminProposal != nil {
 		p := proposal.UpdateAdminProposal
-		err := msg.SetContent(&wasmtypes.UpdateAdminProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			NewAdmin:    p.NewAdmin,
-			Contract:    p.Contract,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on UpdateAdminProposal")
+		sdkMsg = &wasmtypes.MsgUpdateAdmin{
+			Sender:   authority,
+			NewAdmin: p.NewAdmin,
+			Contract: p.Contract,
 		}
 	}
 
 	if proposal.ClearAdminProposal != nil {
 		p := proposal.ClearAdminProposal
-		err := msg.SetContent(&wasmtypes.ClearAdminProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			Contract:    p.Contract,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on ClearAdminProposal")
+		sdkMsg = &wasmtypes.MsgClearAdmin{
+			Sender:   authority,
+			Contract: p.Contract,
+		}
+	}
+
+	if proposal.ParamChangeNewProposal != nil {
+		p := proposal.ParamChangeNewProposal
+		myJsonString := fmt.Sprintf(`{"authority":"%s", "params": "%s"}`, authority, p.NewParams)
```

Comment:
I don't like using raw json string here.
I'm suggesting the following:
* make a field `UpdateParamsMsg` with type `sdk.Msg` in `ParamChangeNewProposal` structure;
* Here we must compare it with `m.AdminKeeper.RegisteredModulesUpdateParams[p.Module].UpdateParamsMsg` to be sure, that the proposed message is a message to update params for a particulare module;
* we must check that `UpdateParamsMsg.Signer()` equals to admin module address (because the module is registered as an authority for other modules);
* if everything is ok, pass the message to the admin-module;

So in that case to create a proposal for changing params in specific module, proposer must compose a proper `UpdateParams` msg for the module (where `authority` is an address of the `admin-module`), create a proposal in Neutron DAO with that message and after it passes, it is being executed and passed to admin-module.

@swelf19 wdyt?